vm/src/air_public_input.rs

```diff
@@ -0,0 +1,122 @@
+use std::collections::HashMap;
+
+use felt::Felt252;
+use serde::Serialize;
+use thiserror::Error;
+
+use crate::{
+    types::layout::CairoLayout,
+    vm::{
+        errors::{trace_errors::TraceError, vm_errors::VirtualMachineError},
+        trace::trace_entry::TraceEntry,
+    },
+};
+
+#[derive(Serialize, Debug)]
+pub struct PublicMemoryEntry {
+    address: usize,
+    page: usize,
+    #[serde(serialize_with = "mem_value_serde::serialize")]
+    value: Option<Felt252>,
+}
+
+mod mem_value_serde {
+    use super::*;
+    use serde::Serializer;
+
+    pub(crate) fn serialize<S: Serializer>(
+        value: &Option<Felt252>,
+        serializer: S,
+    ) -> Result<S::Ok, S::Error> {
+        if let Some(value) = value {
+            serializer.serialize_str(&format!("0x{}", value.to_str_radix(16)))
+        } else {
+            serializer.serialize_none()
+        }
+    }
+}
+
+#[derive(Serialize, Debug)]
+pub struct PublicInput<'a> {
+    layout: &'a str,
+    layout_params: Option<&'a CairoLayout>,
+    rc_min: Option<isize>,
+    rc_max: Option<isize>,
```

Comment:
Does it make sense for these to be `Option`? In what scenario would they be `None`?

Reply:
Done!